app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherPager.kt

```diff
@@ -0,0 +1,192 @@
+
+  val today = LocalDate.now()
+  val eventStartDate = event.startsAt().toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
+  val eventEndDate = event.endsAt().toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
+
+  val targetStartDate = maxOf(today, eventStartDate)
+
+  LaunchedEffect(event) {
+    if (eventEndDate.isBefore(today)) {
+      message = context.getString(R.string.noforecast_alreadyoccured)
+    } else if (targetStartDate.isAfter(today.plusDays(10))) {
+      message = context.getString(R.string.noforecast_eventtoofarinfuture)
+    } else {
+      coroutineScope.launch {
+        val weatherList = mutableListOf<Weather>()
+        val dateRange = generateDateRange(targetStartDate, minOf(eventEndDate, today.plusDays(10)))
+        dateRange.forEach { date ->
+          getWeather(
+              event.location,
+              date,
+              { weather ->
+                weatherList.add(weather)
+                if (weatherList.size == dateRange.size) {
+                  weatherList.sortBy { it.date }
+                  weatherData = weatherList.toList()
+                  Log.d("WeatherApp", "Weather data: $weatherList")
+                }
+              },
+              {
+                // Handle failure
+                message = context.getString(R.string.noforecast_error)
+              },
+              context)
+        }
+      }
+    }
+  }
+
+  Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+    when {
+      message != null -> TextMessage(message)
+      weatherData != null -> WeatherCarouselInternal(weatherData!!)
+      else -> CircularProgressIndicator(modifier = Modifier.testTag("loading_indicator"))
+    }
+  }
+}
+
+fun generateDateRange(start: LocalDate, end: LocalDate): List<LocalDate> {
+  val dates = mutableListOf<LocalDate>()
+  var current = start
+  while (current <= end) {
+    dates.add(current)
+    current = current.plusDays(1)
+  }
+  return dates
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun WeatherCarouselInternal(weatherData: List<Weather>) {
+  val pagerState = rememberPagerState { weatherData.size }
+
+  Column {
+    HorizontalPager(
+        state = pagerState,
+        modifier = Modifier.fillMaxWidth(),
+        beyondBoundsPageCount = weatherData.size - 1) { page ->
+          WeatherCard(weatherData[page])
+        }
+
+    Row(modifier = Modifier.align(Alignment.CenterHorizontally).padding(16.dp)) {
+      repeat(weatherData.size) { iteration ->
+        val color =
+            if (pagerState.currentPage == iteration) MaterialTheme.colorScheme.primary
+            else Color.LightGray
+        Box(modifier = Modifier.padding(2.dp).clip(CircleShape).background(color).size(10.dp))
+      }
+    }
+  }
+}
+
+@Composable
+fun WeatherCard(weather: Weather) {
+  val configuration = LocalConfiguration.current
+  val locale = configuration.locales[0]
+  val formatter = DateTimeFormatter.ofPattern("EEEE dd MMMM yyyy", locale)
```

Comment:
We should add in the PRD something about localisation for dates like this.
Indeed, there are many instances like this in the current codebase, with the date format being not really flexible